roles/smartgateway/templates/deployment.yaml.j2

```diff
@@ -34,7 +34,7 @@ spec:
         - -tls-cert=/etc/tls/private/tls.crt
         - -tls-key=/etc/tls/private/tls.key
         - -cookie-secret-file=/etc/proxy/secrets/session_secret
-        - -openshift-service-account=NA
+        - -openshift-service-account={{ service_account_name }}
```

Comment:
It finally makes sense to me why this worked (rather unexpectedly) - it was using the default account, and the Loki operator adjusted it so that it would work. Loop closed!